src/openrct2/title/TitleSequence.h

```diff
@@ -40,16 +40,64 @@ struct TitleCommand
 
 struct TitleSequence
 {
-    const utf8* Name;
-    const utf8* Path;
+public:
+    const std::string& Path() const
+    {
+        return _path;
+    }
+    const std::string& Name() const
+    {
+        return _name;
+    }
+    bool IsZip() const
+    {
+        return _isZip;
+    }
+    std::vector<TitleCommand>& Commands()
+    {
+        return _commands;
+    }
+    std::vector<std::string>& Saves()
+    {
+        return _saves;
+    }
+    size_t NumSaves() const
+    {
+        return _saves.size();
+    }
+    size_t NumCommands() const
+    {
+        return _commands.size();
+    }
+    void SetName(const std::string& name)
+    {
+        _name = name;
+    }
+    void SetPath(const std::string& path)
+    {
+        _path = path;
+    }
+    void SetIsZip(bool isZip)
+    {
+        _isZip = isZip;
+    }
+    void AcquireCommands(std::vector<TitleCommand>& commands)
+    {
+        _commands = std::move(commands);
+    }
+    void AcquireSaves(std::vector<std::string>& saves)
+    {
+        _saves = std::move(saves);
+    }
 
-    size_t NumCommands;
-    TitleCommand* Commands;
+private:
+    std::string _path;
+    std::string _name;
 
-    size_t NumSaves;
-    utf8** Saves;
+    std::vector<TitleCommand> _commands;
+    std::vector<std::string> _saves;
 
-    bool IsZip;
+    bool _isZip{ false };
 };
```

Comment:
None of this is an invariant therefore i don't see much point making accessors. Just keep all the fields public.

Reply:
He converted the `struct` to a `class`, so it kind of makes sense to make those `private` according to the core guidelines: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-class

Do you suggest also converting it back to a `struct`?

(Personally I like it being a class, specially with all the methods that are about to become member functions)

Reply:
See what it looks like after making more of these member functions. I don't think it needs all of these assessors though.

Reply:
See comment:
https://github.com/OpenRCT2/OpenRCT2/pull/12815#discussion_r481605081